### ISSUE TITLE
Can't open VTMB (Vampire The Masquerade - Bloodlines)

### ISSUE THREAD

**Issue by ghyujkilop** [craftablescience/VPKEdit#164]
### Describe the Bug

![image](https://github.com/user-attachments/assets/ef73d1e1-26bc-4ea9-abaf-5fd613af8b62)

### To Reproduce

I select open, select one of these files:
![image](https://github.com/user-attachments/assets/398d8684-03dc-425f-9e4c-93b5e3e2d816)

Then that error you see above appears. Also, definitely don't have anything using the files. Copied and pasted into a new folder, and nothing uses these.

### Operating System

Windows 11

**Comment by Psycho-A** [comment]
VPK of Bloodlines is totally different format than Valve VPK's. Although of course it would be nice to have native support for this game. But it's more likely to be "Feature request" than "Bug" :)

**Comment by craftablescience** [comment]
Thanks for making this issue when you did, because the game was 50% off and I ended up grabbing it lol

Now that I actually own it I can look into supporting it

**Comment by craftablescience** [comment]
Implemented in backend library in https://github.com/craftablescience/sourcepp/pull/17

**Comment by craftablescience** [comment]
And now implemented in VPKEdit! This feature will be in the next release, thanks for reporting!

**Comment by Psycho-A** [comment]
> And now implemented in VPKEdit! This feature will be in the next release, thanks for reporting!

Wow, you're just a god! I was also really looking forward to Bloodlines support, so as not to have a bunch of specialized inconvenient unpacking utilities. By the way, the VPKs in this game contain random content in each file and are loaded by the game all together at once - so it would be good if VPKEdit loaded all pack***.vpks at once to have a common file system tree.

**Comment by craftablescience** [comment]
> it would be good if VPKEdit loaded all pack***.vpks at once to have a common file system tree.

damn guess i have to do this now lol

technically these files are self-contained, it goes file data -> file tree -> footer, where VPKs are loaded simultaneously because there's one `_dir` file that stores the file tree and a bazillion numbered archives that purely store file data. but i made an exception for RTX Remix PKG files (unmerged, I couldn't wrangle extracting the texture data) and that was a very similar situation to here, so

tl;dr yeah sure why not

**Comment by craftablescience** [comment]
![image](https://github.com/user-attachments/assets/9e271cd0-9dce-4217-b414-5e4bb9107f22)

That seems right

Loading all the VPKs at once will be supported next release

**Comment by Psycho-A** [comment]
It's just great that you even added full-fledged archive editing! But in this case, it would also be good to be able to create completely new archives. I tried it in the new beta version, but the "VPK" section doesn't offer a version for VTMB.

There is also a question: if we open all numbered archives as a single file system, which archive will our new files be added to? I haven't checked it, but I suggest doing this way: if we open the file "pack000.vpk" or "pack100.vpk" - all files will be added to the NEWLY created archive (going next by number) - or just ask the user about which one is use for. But if any other archive is opened (for example, "pack005.vpk"), then open only its contents, and accordingly, add new files to this one only.

**Comment by craftablescience** [comment]
> If we open all numbered archives as a single file system, which archive will our new files be added to?

They will be added to a new numbered archive, which will be `current highest archive number + 1`. I am unsure how their system works, so I elected to open everything rather than the files with consecutive numbers.

**Comment by Psycho-A** [comment]
> They will be added to a new numbered archive, which will be current highest archive number + 1.

So, it's just like I suggested ("all files will be added to the NEWLY created archive, going next by number"), right?

> I am unsure how their system works, so I elected to open everything rather than the files with consecutive numbers.

That's right, although there's just one nuance. Archives starting with "pack0xx" are used in the game for shared resources (textures etc), and the engine reads new archives as "pack0xx + 1" if they're found. The "pack1xx" archives are used for localization files, and if the pack100.vpk file is found in the game folder, the engine also searches for all "pack1xx + 1" files. The formats are completely identical, and in fact, "pack1xx.vpks" are optional - so new files can only be added to "pack0xx + 1". Or might see which archive type is opened by program (pack0xx or pack1xx).

And, what about creation in "File - New" menu?
